src/Compilers/CSharp/Portable/Emitter/EditAndContinue/CSharpSymbolMatcher.cs

```diff
@@ -49,61 +48,60 @@ internal sealed class CSharpSymbolMatcher : SymbolMatcher
                 anonymousTypeMap,
                 sourceAssembly,
                 otherAssembly,
-                otherSynthesizedMembersOpt: null,
-                deepTranslatorOpt: null);
+                otherSynthesizedMembers: null,
+                deepTranslator: null);
         }
 
-        public override Cci.IDefinition MapDefinition(Cci.IDefinition definition)
+        public override Cci.IDefinition? MapDefinition(Cci.IDefinition? definition)
         {
             if (definition?.GetInternalSymbol() is Symbol symbol)
             {
-                return (Cci.IDefinition)_symbols.Visit(symbol)?.GetCciAdapter();
+                return (Cci.IDefinition?)_symbols.Visit(symbol)?.GetCciAdapter();
             }
 
-            return _defs.VisitDef(definition);
+            return null;
```

Comment:
This looks like a break? `_defs` is unused with this change...

Reply:
Seems like this was dead code. If `definition` was null before VisitDef would throw NRE. The other Map* methods all return null in that case, so this change is doing the same. 

---
In reply to: [585215246](https://github.com/dotnet/roslyn/pull/51576#discussion_r585215246) [](ancestors = 585215246)

Reply:
Following up with some additional cleanup. The parameters seem to be non-nullable at all call sites.

---
In reply to: [585886656](https://github.com/dotnet/roslyn/pull/51576#discussion_r585886656) [](ancestors = 585886656,585215246)

Reply:
Interesting that _defs is never used. Odd. Need to dig into git history a bit.

---
In reply to: [585891716](https://github.com/dotnet/roslyn/pull/51576#discussion_r585891716) [](ancestors = 585891716,585886656,585215246)

Reply:
The condition is 7 years old. I'll put it back and file an issue to follow up. It might all be dead code and we should simplify this.

---
In reply to: [585899490](https://github.com/dotnet/roslyn/pull/51576#discussion_r585899490) [](ancestors = 585899490,585891716,585886656,585215246)